### PR TITLE
rust-> palindrome program, enforces specification in #108

### PR DESCRIPTION
**Pull request by gauthamkrishna9991**
This is an example specification which is worked into a real-world palindrome problem in `strings/rust/`. This is inside a `palindrome` directory which holds the project files.

## Directory Structure
```bash
palindrome/   # directory of `palindrome` problem
| .gitignore  # gitignore for folder/directory
| Cargo.lock  # spec file, machine-readable lock file
| Cargo.toml  # spec file, human-readable
| README.md   # very important README file, describes specifications + instructions
| src         # source directory
  | main.rs   # main(entry point) file
```
The reason why the `Cargo.lock` file was also included, [read here](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries).

## `README.md` file

This contains the `README` for the specific problem.

Structure of `README.md`
```markdown
`project_name`

Language: <Language Name> ( `language_code` )

Main/Entry File: [relative/path/to/file](relative/path/to/file)

Algorithm File: [relative/path/to/file](relative/path/to/file) <if needed>

Author: [Author_Name](github.com/link-to-author-profile)

---

# A Readable Title

< description >

## Prerequisites

- list
- of
- prerequisites (https://example.com/link-if-needed)

## Build Instruction

- Readable and 
- Reproducible Instructions
- If Needed
- If contains a terminal command, show it like this
    ```bash
     # put build command here
    ```
## Test Cases & Output <if relevant>
- `list`
- `of`
- `test`
- `cases` < -> `output if needed` >

```
Check the `README.md` (`strings/rust/palindrome/README.md`) file to understand it better, as an example.

This is an example for #108.

### CODE REVIEW

**Review by Arsenic-ATG — COMMENTED** [review]
looks good to me

**Comment by gauthamkrishna9991** [comment]
If it's fine with all of you, I would add one more commit, which would add the `Rust` category in the `strings` README file.

File: strings/rust/palindrome/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Language: Rust ( `rust` )
+
+Main/Entry File: [`src/main.rs`](src/main.rs)
+
+Algorithm File: [`src/main.rs`](src/main.rs)
+
+Author: [Goutham Krishna](https://github.com/gauthamkrishna9991/)
+
+---
+
```
Comment (ming-tsai) [inline strings/rust/palindrome/README.md:13]: Hi @gauthamkrishna9991,
Please delete theses lines, you not need them, due to we are using git

**Review by ming-tsai — CHANGES_REQUESTED** [review]

File: strings/rust/palindrome/README.md (hunk 1 above)
Comment (gauthamkrishna9991) [reply strings/rust/palindrome/README.md]: Done.

**Review by TawfikYasser — COMMENTED** [review]
Good @gauthamkrishna9991

**Review by ming-tsai — APPROVED** [review]
Nice work @gauthamkrishna9991

**Pull request merged**